User guide spell checks and replaced http with https links

Adriani90 opened the pull request:
### Link to issue number:
fixes #9508 
fixes #8314

### Summary of the issue:
* There are some examples missing in the user guide (i.e. regarding object description)and some text passages should be deleted (i.e. that eSpeak is the only synthesizer which suports infelction and change in pitch or automatic language switch). Also there are some miss spellings. Most links are still http: and not https:
* Some links for braille displays are outdated (i.e. baum and Seika).

### Description of how this pull request fixes the issue:
• Edited the user guide accordingly and fixed the spelling errors. In the buildVersionInfo, the URL is a https: now as well as all other links in the user guide.
• Links for Seika and Baum braille displays are up to date.

### Testing performed:
Ran NVDA from source and checked that formating is not changed and that links are working.

### Known issues with pull request:
none

### Change log entry:
none

Adriani90 committed:
Replaced http with https in links, solved some spell checks, removed "supported only by eSpeak" for inflection and change in pitch since this is also supported by other synthesizers.

Adriani90 committed:
Completed replacing http with https, added a current reference for Seika braille display and changed the baum website to visiobraille which is the new product owner of Baum products.

Adriani90 committed:
Merge branch 'master' of https://github.com/nvaccess/nvda into User_guide_spell_checks_and_https_links

Adriani90 committed:
splited some sentences into two lines

Adriani90 committed:
Splited some text into two lines

Adriani90 commented:
Hey Reef, so this is the most comprehensive one I guess. I do it now by cherrypicking the commits. It's my first one and I didn't know that it causes so much work. Are you able to review this one?

feerrenrut commented on user_docs/en/userGuide.t2t:
```diff
@@ -1115,7 +1115,7 @@ For no pitch change you would use 0.
 ==== Say "cap" before capitals ====[SpeechSettingsSayCapBefore]
 This setting is a checkbox that, when checked, tells NVDA to say the word "cap" before any capital letter when spoken as an individual character such as when spelling.
-Usually, NVDA raises the pitch slightly for any capital letter, but some synthesizers may not support this well, so perhaps this option may be of use.
+Usually, NVDA raises the pitch slightly for any capital letter, but some synthesizers may not support this well, so perhaps this option may be preferred alternative for some users.
```
On line 1122 (under "Beep for capitals")

Instead of:
> Like the "say cap for capitals" checkbox, this is useful for Synthesizers that can't change their pitch for capital letters.

Use the same sentence:
> This option is also useful for Synthesizers that can't change their pitch for capital letters.

feerrenrut commented on user_docs/en/userGuide.t2t:
```diff
@@ -1617,7 +1619,7 @@ You can configure reporting of:
 
 ==== Report formatting changes after the cursor ====[DocumentFormattingDetectFormatAfterCursor]
-If enabled, this setting tells NVDA to try and detect all the formatting changes on a line as it reports it, even if doing this may slow down NVDA's performance.
+If enabled, this setting tells NVDA to try and detect all the formatting changes on a line as it reports it - even if doing this may slow down NVDA's performance.
```
This seems a little bit strange to me, I would expect a comma here or perhaps a semi-colon.

Also for the change in restoring the default settings below this.

feerrenrut commented on user_docs/en/userGuide.t2t:
```diff
@@ -1665,8 +1667,8 @@ This button opens the directory where you can place custom code while developing
 ==== Use UI automation to access Microsoft Word document controls when available ====[AdvancedSettingsUseUiaForWord]
-When this option is enabled, NVDA will try to use  the Microsoft UI Automation accessibility api in order to fetch information from Microsoft Word document controls.
-This includes in Microsoft Word itself, and also the Microsoft Outlook message viewer and composer.
+When this option is enabled, NVDA will try to use  the Microsoft UI Automation accessibility API in order to fetch information from Microsoft Word document controls.
```
There is a double space after "NVDA will try to use" might as well fix it while changing this line.

feerrenrut reviewed (changes requested):
Please confirm that you have tested all the links that you have changed.

feerrenrut commented:
Sorry, just re-read the description and noted that you have tested the links.

Adriani90 committed:
Addressed review actions

Adriani90 commented:
Reef, I have addressed your suggestions. For say "cap" before and beeps for caps, I have just added a sentence under change pitch percentage which says that the two alternative options are helpful if the synth does not suport pitch change for caps. So in my view we do not have to say this in the both alternative options. The users will already understand it if they read "change pitch percentage".

Adriani90 commented:
thanks very much for your review.

feerrenrut commented on user_docs/en/userGuide.t2t:
```diff
@@ -70,7 +70,7 @@ For further details, you can [view the full licence. http://www.gnu.org/licenses
 
 + Getting and Setting Up NVDA +[GettingAndSettingUpNVDA]
-If you have not yet got a copy of NVDA, you can download it from [www.nvaccess.org NVDA_URL].
+If you have not yet got a copy of NVDA, you can download it from [NVAccess website NVDA_URL].
```
A "the" is missing. Also note that there should be a space between NV and Access: "from the NV Access website." 

```suggestion
If you have not yet got a copy of NVDA, you can download it from the [NVAccess website NVDA_URL].
```

Adriani90 committed:
Merge branch 'master' of https://github.com/nvaccess/nvda into User_guide_spell_checks_and_https_links

Adriani90 committed:
Update user_docs/en/userGuide.t2t

Co-Authored-By: Reef Turner <feerrenrut@users.noreply.github.com>

Adriani90 committed:
added space

Adriani90 commented:
so now I think this should be ready.

Adriani90 commented:
Reef would it be ok if I fix #6211 also in this PR? It is such a small change that is not worthy to raise a new PR for. I just added two missing "the" in the input help message for say all command. So is it ok if I commit the cahnge here and request a new review for that?

Adriani90 committed:
Small fix in the input help message for say all command

Adriani90 committed:
Merge branch 'master' of https://github.com/nvaccess/nvda into User_guide_spell_checks_and_https_links

Adriani90 commented:
@feerrenrut I get flake8 errors when merging NV Access's master in to my branch with the related commits. Could you please take a look at this?

Adriani90 committed:
Merge branch 'master' of https://github.com/nvaccess/nvda into User_guide_spell_checks_and_https_links

Adriani90 commented:
What should we do with E501 and E225? White space around operators is an estetical thing and this is not given in all cases in versionsinfo.
Regarding the line being too long, this is also something which could be ignored by default. There are lots of lines in global commands which are longer than 110 characters.
Should I solve the lint errors for this PR despite the inconsistency which would result from it?
cc: @leonardder

feerrenrut commented:
@Adriani90 Yes, please fix the lint errors. This will be inconsistent with the other lines in the file, but over time we will be working towards greater consistency across the whole code base. 

If there is a good reason for it, you can stop a lint error from appearing by adding a comment to the line like ` # noqa: E501`. I don't expect either of the cases in this PR require an exception to be made, but if you ever do need to make an exception, please add a description of the warning number and a good explanation of why this exception is in place.

Adriani90 committed:
Fixed E225 lint error in versionInfo.py

Adriani90 committed:
Fixed E501 lint error

Adriani90 commented:
Hmm, the checks still fail and I don't really understand why. :-) VersionInfo.py has spaces around operators and the input help message for review cursor is now splited into two lines, 102 and 48 characters respectively, both below 110 characters as requested.

feerrenrut commented:
The build actually failed, not the linter. It was not able to checkout a submodule. You can check by going to "show all checks" then "details" and reading the console output for the build.

Looking at the diff for this, as a result of merging this there will be changes to various submodules. You can try rebasing this on master or it might be easier to manually re-apply your changes to a fresh checkout of master. 

Note, you can run the linter locally, if this passes, it should on the PR as well.

Adriani90 commented:
Closing in favor of #10130
